client/clients/router/client.go

```diff
@@ -92,3 +138,388 @@ type Client interface {
 	// The returned regions are flattened, even there are key ranges located in the same region, only one region will be returned.
 	BatchScanRegions(ctx context.Context, keyRanges []KeyRange, limit int, opts ...opt.GetRegionOption) ([]*Region, error)
 }
+
+// Cli is the implementation of the router client.
+type Cli struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+	option *opt.Option
+
+	svcDiscovery sd.ServiceDiscovery
+	// leaderURL is the URL of the router leader.
+	leaderURL atomic.Value
+	// conCtxMgr is used to store the context of the router stream connection(s).
+	conCtxMgr *cctx.Manager[pdpb.PD_QueryRegionClient]
+	// updateConnectionCh is used to trigger the connection update actively.
```

Comment:
Should updateConnectionCh be part of conCtxMgr?

Reply:
The client controls connection updating with the URL change, and `conCtxMgr` works more like a storage layer for the connections. It seems there is no particular reason to put the update notifications in there.